ksqldb-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -181,8 +197,15 @@ protected void uncaughtHandler(final Thread t, final Throwable e) {
           );
       queryStateListener.ifPresent(lis -> lis.onError(queryError));
       queryErrors.add(queryError);
-      LOG.error("Unhandled exception caught in streams thread {}. ({})", t.getName(), errorType, e);
+      LOG.error(
```

Comment:
Please see the other comment above: now we are only adding queryError to the queue of QueryMetadata, but does not clean it once it resumes back to normal (since it does not go through the fail and resume pattern any more). The errors would just be an LRU cache. Is that by design?

Reply:
+1 I think this is an issue. The way I was thinking we could solve this would be to make the queryErrors queue a time-bounded queue. So we can have the entries automatically expired once they've been sitting in the queue longer than some time threshold (lets say 1 hour to start)

Reply:
I made a time based queue wrapping around the eviction queue as I could not find a time based eviction queue. if you had some implementation in mind let me know and I can replace it.